### PR TITLE
New roles permissions

### PR DESCRIPTION
**Pull request by venkylads**
done few changes and also seeder is added for JobOpportunity model and readme file is included

### CODE REVIEW

File: online-dashboard-api/app/Http/Controllers/JobOpportunityController.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * Show the form for creating a new resource.
+     */
+    public function create() {}
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(CreateJObRequest $request): JsonResponse
+    {
+        JobOpportunity::createJob($request);
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Http/Controllers/JobOpportunityController.php:35]: JobOpportunity::create($request);

**Comment by AKHIL-882** [comment]
Modify all the messages in Title Case
Ex: admin and user roles created==> Admin and User Roles Created

File: online-dashboard-api/app/Http/Requests/CreateJObRequest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Http/Requests/CreateJObRequest.php:1]: File Name => CreateJobRequest.php

File: online-dashboard-api/app/Http/Requests/UpdateJobRequest.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Http/Requests/UpdateJobRequest.php:1]: Why to have same data in two request : CreateJobRequest and UpdateJobRequest
Use single Request as JobRequest, that would be fine,

File: online-dashboard-api/app/Models/User.php [hunk 4]
```diff
@@ -76,10 +77,14 @@ public function initiatePasswordReset(): void
...
     public static function createUser($data): object
     {
-        return self::create([
+        $user = self::create([
             'name' => $data['name'],
             'email' => $data['email'],
             'password' => Hash::make($data['password']),
         ]);
+
+        $user->assignRole('user');
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Models/User.php:86]: Where the assignRole() functions or did I miss anything!?

File: online-dashboard-api/database/migrations/2024_12_24_073310_create_job_opportunities_table.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            $table->timestamps();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('add_jobs');
```
Comment (AKHIL-882) [inline online-dashboard-api/database/migrations/2024_12_24_073310_create_job_opportunities_table.php:30]: Schema::dropIfExists('job_opportunities');

File: online-dashboard-api/app/Models/User.php (hunk 4 above)
Comment (venkylads) [reply online-dashboard-api/app/Models/User.php:86]: i didn't get you, but it's HasRoles trait function from spatie pacakage you just need to import the trait in you model then yoiu can use it whenever and whereever you want.

File: online-dashboard-api/app/Http/Requests/UpdateJobRequest.php (hunk 3 above)
Comment (venkylads) [reply online-dashboard-api/app/Http/Requests/UpdateJobRequest.php:1]: done

File: online-dashboard-api/database/migrations/2024_12_24_073310_create_job_opportunities_table.php (hunk 5 above)
Comment (venkylads) [reply online-dashboard-api/database/migrations/2024_12_24_073310_create_job_opportunities_table.php:30]: done

File: online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class JobOpportuinityRequest extends FormRequest
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php:7]: class name! 😢

File: online-dashboard-api/app/Http/Resources/JobResource.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
Comment (AKHIL-882) [inline online-dashboard-api/app/Http/Resources/JobResource.php:2]: You can delete this file and use the JobOpportunityRequest for all

File: online-dashboard-api/app/Models/User.php (hunk 4 above)
Comment (AKHIL-882) [reply online-dashboard-api/app/Models/User.php:86]: Okay! assignRole() this function is driving from the library isn't it! ?

File: online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php (hunk 6 above)
Comment (venkylads) [reply online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php:7]: i didn't get you

File: online-dashboard-api/app/Models/User.php (hunk 4 above)
Comment (venkylads) [reply online-dashboard-api/app/Models/User.php:86]: yup

File: online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php (hunk 6 above)
Comment (venkylads) [reply online-dashboard-api/app/Http/Requests/JobOpportuinityRequest.php]: done

**Review by AKHIL-882 — APPROVED** [review]

**Pull request merged**